### PR TITLE
Return Initialize hint from most things that don't

### PR DESCRIPTION
**Pull request by moxian**
## What Does This PR Do
Changes
```
/thing/Initialize(mapload)
    ..()
```
to 
```
/thing/Initialize(mapload)
    . = ..()
```
Changes performed via find-and-replace (with a tiny bit of regex magic) + a manual update to the handful of leftover bits

## Why It's Good For The Game
SSatoms sorta expects an explicit initialize hint to be returned. It doesn't outright *break* if it's not, but it does get confused and complain.

I'm not sure how to put it, but this is part of #14770 revival effort. The other part of it has something to do with ruins and/or white ship, but i'm not going to fix that right now.

### CODE REVIEW

File: code/game/gamemodes/miniantags/abduction/machinery/console.dm [hunk 1]
```diff
@@ -29,7 +29,7 @@
 	var/list/datum/icon_snapshot/disguises = list()
 
 /obj/machinery/abductor/console/Initialize()
-	..()
+	. = ..()
 	return INITIALIZE_HINT_LATELOAD
```
Comment (AffectedArc07) [inline code/game/gamemodes/miniantags/abduction/machinery/console.dm:33]: This is why you dont do mass find/replace.

This call was fine, as it returns a hint right underneath it, and the setting of `.` incurs extra instruction overhead.

File: code/game/machinery/computer/depot.dm [hunk 2]
```diff
@@ -399,7 +399,7 @@
 	var/portaldir = WEST
 
 /obj/machinery/computer/syndicate_depot/teleporter/Initialize(mapload)
-	..()
+	. = ..()
 	return INITIALIZE_HINT_LATELOAD
```
Comment (AffectedArc07) [inline code/game/machinery/computer/depot.dm:403]: Ditto

File: code/game/machinery/computer/message.dm [hunk 3]
```diff
@@ -35,7 +35,7 @@
 	light_color = LIGHT_COLOR_DARKGREEN
 
 /obj/machinery/computer/message_monitor/Initialize()
-	..()
+	. = ..()
 	return INITIALIZE_HINT_LATELOAD // Give the message server time to initialize
```
Comment (AffectedArc07) [inline code/game/machinery/computer/message.dm:39]: Ditto

File: code/game/machinery/computer/message.dm [hunk 4]
```diff
@@ -502,7 +502,7 @@
 	var/obj/machinery/message_server/server = null
 
 /obj/item/paper/monitorkey/Initialize(mapload)
-	..()
+	. = ..()
 	return INITIALIZE_HINT_LATELOAD
```
Comment (AffectedArc07) [inline code/game/machinery/computer/message.dm:506]: And again

File: code/game/machinery/doors/brigdoors.dm [hunk 5]
```diff
@@ -115,7 +115,7 @@
 	atom_say("[src] beeps, \"[occupant]: [notifytext]\"")
 
 /obj/machinery/door_timer/Initialize(mapload)
-	..()
+	. = ..()
```
Comment (AffectedArc07) [inline code/game/machinery/doors/brigdoors.dm:118]: Same here

File: code/game/machinery/lightswitch.dm [hunk 6]
```diff
@@ -47,7 +47,7 @@
 		update_icon(UPDATE_ICON_STATE)
 
 /obj/machinery/light_switch/Initialize()
-	..()
+	. = ..()
 	set_frequency(frequency)
 	name = "light switch"
```
Comment (AffectedArc07) [inline code/game/machinery/lightswitch.dm:52]: Why the bloody hell is this here?
*also good god I just looked at the above proc*

File: code/game/machinery/status_display.dm [hunk 7]
```diff
@@ -56,7 +56,7 @@
 
 // register for radio system
 /obj/machinery/status_display/Initialize()
-	..()
+	. = ..()
 	if(SSradio)
 		SSradio.add_object(src, frequency)
 	update_icon()
```
Comment (AffectedArc07) [inline code/game/machinery/status_display.dm:62]:
```suggestion
	SSradio.add_object(src, frequency)
	update_icon()
```
`SSradio` always exists.

File: code/game/machinery/tcomms/_base.dm [hunk 8]
```diff
@@ -434,7 +434,7 @@ GLOBAL_LIST_EMPTY(tcomms_machines)
   * Otherwise shit breaks BADLY
   */
 /obj/item/paper/tcommskey/Initialize(mapload)
-	..()
+	. = ..()
 	return INITIALIZE_HINT_LATELOAD
```
Comment (AffectedArc07) [inline code/game/machinery/tcomms/_base.dm:438]: And again

File: code/game/machinery/turret_control.dm [hunk 9]
```diff
@@ -72,7 +72,7 @@
 	return ..()
 
 /obj/machinery/turretid/Initialize()
-	..()
+	. = ..()
```
Comment (AffectedArc07) [inline code/game/machinery/turret_control.dm:75]: Could you remove the redundant return from the end of this proc

File: code/game/objects/effects/gibs.dm [hunk 10]
```diff
@@ -17,7 +17,7 @@
 	var/list/gibdirections = list() //of lists
 
 /obj/effect/gibspawner/Initialize(mapload, datum/dna/mob_dna)
-	..()
+	. = ..()
```
Comment (AffectedArc07) [inline code/game/objects/effects/gibs.dm:20]: And again, hint is set on 27

File: code/game/objects/effects/spawners/airlock_spawner.dm [hunk 11]
```diff
@@ -40,7 +40,7 @@ This spawner places pipe leading up to the interior door, you will need to finis
 	var/one_door_exterior //See above
 
 /obj/effect/spawner/airlock/Initialize()
-	..()
+	. = ..()
```
Comment (AffectedArc07) [inline code/game/objects/effects/spawners/airlock_spawner.dm:43]: Remove the dot-set here, and replace `qdel(src)` with the qdel hint.

File: code/game/objects/items/random_items.dm [hunk 12]
```diff
@@ -5,7 +5,7 @@
 	name = "Random Toy"
 
 /obj/item/toy/random/Initialize(mapload)
-	..()
+	. = ..()
```
Comment (AffectedArc07) [inline code/game/objects/items/random_items.dm:8]: And again

File: code/game/objects/items/weapons/dna_injector.dm [hunk 13]
```diff
@@ -158,7 +158,7 @@
 
 /obj/item/dnainjector/hulkmut/Initialize()
 	block = GLOB.hulkblock
-	..()
+	. = ..()
```
Comment (AffectedArc07) [inline code/game/objects/items/weapons/dna_injector.dm:161]: Might be out of the scope of this PR, but could you define a `getBlock()` proc for injectors and have the parent call use that akin to `populateContents()` on a closet?

The fact the super call is second here instead of first is killing me.

File: code/game/objects/items/weapons/manuals.dm [hunk 14]
```diff
@@ -891,7 +891,7 @@
 	var/static/list/banned_books = list(/obj/item/book/manual/random, /obj/item/book/manual/nuclear, /obj/item/book/manual/wiki)
 
 /obj/item/book/manual/random/Initialize()
-	..()
+	. = ..()
 	var/newtype = pick(subtypesof(/obj/item/book/manual) - banned_books)
 	new newtype(loc)
 	return INITIALIZE_HINT_QDEL
```
Comment (AffectedArc07) [inline code/game/objects/items/weapons/manuals.dm:897]: And again

File: code/game/objects/structures/safe.dm [hunk 15]
```diff
@@ -393,7 +393,7 @@ GLOBAL_LIST_EMPTY(safes)
 	info = "<div style='text-align:center;'><img src='ntlogo.png'><center><h3>Safe Codes</h3></center>"
 
 /obj/item/paper/safe_code/Initialize(mapload)
-	..()
+	. = ..()
 	return INITIALIZE_HINT_LATELOAD
```
Comment (AffectedArc07) [inline code/game/objects/structures/safe.dm:397]: And again.

File: code/modules/assembly/signaler.dm [hunk 16]
```diff
@@ -24,7 +24,7 @@
 		set_frequency(frequency)
 
 /obj/item/assembly/signaler/Initialize()
-	..()
+	. = ..()
 	if(SSradio)
 		set_frequency(frequency)
```
Comment (AffectedArc07) [inline code/modules/assembly/signaler.dm:29]:
```suggestion
	set_frequency(frequency)
```

File: code/modules/awaymissions/loot.dm [hunk 17]
```diff
@@ -6,7 +6,7 @@
 	var/loot = ""			//a list of possible items to spawn- a string of paths
 
 /obj/effect/spawner/away/lootdrop/Initialize()
-	..()
+	. = ..()
 	var/list/things = params2list(loot)
 
 	if(things && things.len)
```
Comment (AffectedArc07) [inline code/modules/awaymissions/loot.dm:12]:
```suggestion
	..()
	var/list/things = params2list(loot)

	if(length(things))
```

And replace `qdel(src)` on 25 with the appropriate hint.

File: code/modules/mapping/mapping_helpers.dm [hunk 18]
```diff
@@ -60,7 +60,7 @@
 	var/late = FALSE
 
 /obj/effect/mapping_helpers/Initialize(mapload)
-	..()
+	. = ..()
 	return late ? INITIALIZE_HINT_LATELOAD : qdel(src) // INITIALIZE_HINT_QDEL <-- Doesn't work
```
Comment (AffectedArc07) [inline code/modules/mapping/mapping_helpers.dm:64]: Ditto.

Also can you check the claim in that comment?

File: code/modules/power/turbine.dm [hunk 19]
```diff
@@ -344,7 +344,7 @@
 
 
 /obj/machinery/computer/turbine_computer/Initialize()
-	..()
+	. = ..()
 	spawn(10)
 		locate_machinery()
```
Comment (AffectedArc07) [inline code/modules/power/turbine.dm:349]: Remove this spawn. This is a carryover from when this was still `/New()` and the atoms didnt exist physically in the world. 

Calling the locate on init should be safe now.

File: code/modules/research/experimentor.dm [hunk 20]
```diff
@@ -65,7 +65,7 @@
 				critical_items += I
 
 /obj/machinery/r_n_d/experimentor/Initialize(mapload)
-	..()
+	. = ..()
```
Comment (AffectedArc07) [inline code/modules/research/experimentor.dm:68]: Again.

File: code/modules/research/xenobiology/xenobio_camera.dm [hunk 21]
```diff
@@ -42,7 +42,7 @@
 	icon_keyboard = "rd_key"
 
 /obj/machinery/computer/camera_advanced/xenobio/Initialize(mapload)
-	..()
+	. = ..()
 	return INITIALIZE_HINT_LATELOAD
```
Comment (AffectedArc07) [inline code/modules/research/xenobiology/xenobio_camera.dm:46]: *And again*

File: code/modules/ruins/lavalandruin_code/puzzle.dm [hunk 22]
```diff
@@ -36,7 +36,7 @@
 			return get_step(center,SOUTHEAST)
 
 /obj/effect/sliding_puzzle/Initialize(mapload)
-	..()
+	. = ..()
 	return INITIALIZE_HINT_LATELOAD
```
Comment (AffectedArc07) [inline code/modules/ruins/lavalandruin_code/puzzle.dm:40]: **And again**

File: code/modules/shuttle/emergency.dm [hunk 23]
```diff
@@ -363,7 +363,7 @@
 	var/target_area = /area/mine/unexplored
 
 /obj/docking_port/stationary/random/Initialize()
-	..()
+	. = ..()
 	var/list/turfs = get_area_turfs(target_area)
 	var/turf/T = pick(turfs)
 	src.loc = T
```
Comment (AffectedArc07) [inline code/modules/shuttle/emergency.dm:369]: Implied `src`
```suggestion
	loc = T
```

**Review by AffectedArc07 — CHANGES_REQUESTED** [review]
90% of the review here could have been avoided if you manually did the change instead of just going gung-ho with regex mass replace.

Theres a reason I dislike people doing it, and this is a perfect example.

File: code/game/objects/items/weapons/dna_injector.dm (hunk 13 above)
Comment (moxian) [reply code/game/objects/items/weapons/dna_injector.dm]: Done

File: code/game/machinery/lightswitch.dm (hunk 6 above)
Comment (moxian) [reply code/game/machinery/lightswitch.dm:52]: I claim "out of scope" on this

File: code/modules/mapping/mapping_helpers.dm (hunk 18 above)
Comment (moxian) [reply code/modules/mapping/mapping_helpers.dm]: Checked.
This is a mess.
TL;DR: the claim is wrong.
TS;PE:
There are two direct children of `/obj/effect/mapping_helpers/` - `/airlock`, and `/no_lava`.
`airlock` has `var/late` set to `TRUE` so the parent returns `INITIALIZE_HINT_LATELOAD`. Then `airlock/Initialize()` promptly ignores it and returns its own `INITIALIZE_HINT_QDEL` unconditionally. It doesn't have LateInitialize either.

`no_lava` doesn't even use Initialize - it's a plain `New`.

The qdel works just fine on both of those.

File: code/modules/power/turbine.dm (hunk 19 above)
Comment (moxian) [reply code/modules/power/turbine.dm]: Done

**Comment by moxian** [comment]
I'd like to apologize for subjecting you to this.  Tbh you should have called me out the first moment you saw bullshit in the PR and not gone through the rest of it.

Anyhow, reviews addressed. PTAL.

**Comment by SteelSlayer** [comment]
Closing this because the author is not currently around to address any changes and/or fix conflicts at this time.

image

Please feel free to re-open this PR should you start contributing to Paradise again.
